torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -430,19 +481,33 @@ class GpuLower::KernelIrMapper : private OptInConstDispatch {
     TORCH_CHECK(gpu_lower_->kir_val_map_.insert({node, lowered_node}).second);
   }
 
+  template <typename KirT, typename FirT>
+  void handleInputOrScalar(const FirT* node) {
+    if (node->isFusionInput()) {
+      // For fusion scalar input nodes that are arguments we create
+      // kir::NamedScalars, as those represent runtime substituted value.
+      const auto lowered_node = ir_builder_.create<kir::NamedScalar>(
+
+          std::string("arg") + KirT::scalarTypePrefix() +
+              std::to_string(node->name()),
+          node->getDataType().value());
```

Comment:
Have we decided to do this? Was there some concerns? Pinging @shmsong

Reply:
If we're not I can also prepare version without it.

Reply:
Should I prepare version without this change comparison?

Reply:
I think it should be ok if only scalar input is named and the scalar's name is automatically generated. We'll probably need to keep track of the created names if we keeping creating different kinds of `NamedScalar`'s later. 

@prak-nv could you show an example of what the generated kernel (maybe just a snippet) looks like with this change?

Reply:
Will paste it morning.

Reply:
I am struggling right now to get my exact case (int argument dependend kir::Allocation) after rebase, but it is basically how it looks like:
`[ RUN      ] NVFuserTest.FusionScalarInputs_CUDA

KERNEL (T0, T1, argd8, argd9, argd10, argd11) -> (T4) :
  DEFINE_MAGIC_ZERO
  d12 = ALLOCATE(mem_type=register, size=1, zero_init=false)
  d12 = argd8 * argd9
  d13 = ALLOCATE(mem_type=register, size=1, zero_init=false)
  d13 = argd10 - argd11
  ~ i24 = T0.size[0] * T0.size[1]
  ~ i26 = ceilDiv(i24, 128)
  ~ i29 = ceilDiv(i26, 4)
  FOR ki96 in id18[blockIdx.x.i(0 .. i29)]:
`
Will try to catch it later with sophisticated enough asserts.

Reply:
After:
```
[ RUN      ] NVFuserTest.FusionSmemDynamicPersistentSoftmax2D_CUDA

KERNEL (T0, argi10) -> (T7) :
  ~ i14 = ceilDiv(T0.size[1], argi10)
  ~ ki142 = i14 * argi10
  T8 = ALLOCATE(mem_type=shared, size=ki142, zero_init=false)
  ~ i23 = ceilDiv(T0.size[1], argi10)
  ~ ki146 = i23 * argi10
  T4 = ALLOCATE(mem_type=shared, size=ki146, zero_init=false)
      .alias=T8
```
Before:
```
[ RUN      ] NVFuserTest.FusionSmemDynamicPersistentSoftmax2D_CUDA

KERNEL (T0, i10) -> (T7) :
  ~ i14 = ceilDiv(T0.size[1], i10)
  ~ ki142 = i14 * i10
  T8 = ALLOCATE(mem_type=shared, size=ki142, zero_init=false)
  ~ i23 = ceilDiv(T0.size[1], i10)
  ~ ki146 = i23 * i10
  T4 = ALLOCATE(mem_type=shared, size=ki146, zero_init=false)
      .alias=T8
```